Description: Made code compatible with GCC 15.
 Fixes this compiler error:

  libyasm/bitvect.h:86:32: error: cannot use keyword 'false' as enumeration constant
     86 |         typedef enum boolean { false = FALSE, true = TRUE } boolean;
        |                                ^~~~~
  libyasm/bitvect.h:86:32: note: 'false' is a keyword with '-std=c23' onwards
Author: Martin 'JaMa' Jansa
Origin: https://github.com/yasm/yasm/pull/287
Bug: https://github.com/yasm/yasm/issues/283
Bug-Debian: https://bugs.debian.org/1098187
Forwarded: not-needed
Last-Update: 2025-04-28
---
--- yasm-1.3.0.orig/libyasm/bitvect.h
+++ yasm-1.3.0/libyasm/bitvect.h
@@ -83,7 +83,11 @@ typedef  Z_longword         *Z_longwordp
     #ifdef MACOS_TRADITIONAL
         #define boolean Boolean
     #else
-        typedef enum boolean { false = FALSE, true = TRUE } boolean;
+        #if __STDC_VERSION__ < 202311L
+             typedef enum boolean { false = FALSE, true = TRUE } boolean;
+        #else
+             typedef bool boolean;
+        #endif
     #endif
 #endif
 
